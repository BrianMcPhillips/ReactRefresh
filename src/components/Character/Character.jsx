import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image }) => (
  <div>
    <img src={image} alt={name}/>
  </div>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Character;
