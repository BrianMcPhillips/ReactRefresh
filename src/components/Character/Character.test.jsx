import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Character 
      name="Test"
      image="http://placecage.com/200/200"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
