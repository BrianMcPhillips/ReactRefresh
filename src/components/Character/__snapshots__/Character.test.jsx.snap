// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/detail/124532"
    >
      <img
        alt="Test"
        src="http://placecage.com/200/200"
      />
    </a>
    <h2>
      Test
    </h2>
  </div>
</DocumentFragment>
`;
